### PR TITLE
release checklist

### LINKED ISSUE #868: Document release process checklist
- Document on each release with migration change how to run the migrate tool and if lakeFS required to stop while the process is running.

### PR DESCRIPTION
**Pull request by nopcoder**
Closes #868

### CODE REVIEW

**Review by guy-har — APPROVED** [review]

**Comment by nopcoder** [comment]
Please review and comment on the release checklist for lakeFS

File: release-checklist.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Release Checklist
+
+- [ ] [Master branch checks](https://github.com/treeverse/lakeFS/actions?query=branch%3Amaster) are green 
+- [ ] Create a [new draft release](https://github.com/treeverse/lakeFS/releases/new)
+    - Bump the version based on SemVer guidelines [SemVer](https://semver.org/)
+    - Make sure that version starts with 'v' and the semver number
+    - Database migration is part of new feature and not a patch
```
Comment (johnnyaug) [inline release-checklist.md:7]:
```suggestion
- [ ] Create a [new draft release](https://github.com/treeverse/lakeFS/releases/new)
    - Bump the version based on [SemVer](https://semver.org/) guidelines.
    - Make sure the version is `v` succeeded by the SemVer number. Example: `v1.1.14`.
    - Database migration is part of new feature and not a patch
```

File: release-checklist.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+- [ ] Verify that [GoReleaser actions](https://github.com/treeverse/lakeFS/actions?query=workflow%3Agoreleaser) triggered by new draft release complates successfully
```
Comment (johnnyaug) [inline release-checklist.md:8]:
```suggestion
- [ ] Verify that the [GoReleaser actions](https://github.com/treeverse/lakeFS/actions?query=workflow%3Agoreleaser) triggered by the new draft release completes successfully.
```

File: release-checklist.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+- [ ] Update the release notes (after goreleaser auto-generated list)
+    - Remove test and documentation related stuff
+    - Try to sort the list by importance
+    - Update the description of items if needed so the reader will understand better what this release changes
+    - Migrateion should be noted and documented outside the changelog (check the changelog for migrate: prefix)
```
Comment (johnnyaug) [inline release-checklist.md:13]:
```suggestion
    - Migration should be noted and documented outside the changelog (check the changelog for migrate: prefix):
```

File: release-checklist.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    - How to run the migrate if needed
+    - Is lakeFS service needs to be down while migrate is running (backward compatibility)
```
Comment (johnnyaug) [inline release-checklist.md:15]:
```suggestion
    - How to run the migration if needed
    - Whether lakeFS service needs to be down while migration is running (backward compatibility)
```

File: release-checklist.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+- [ ] Uncheck the 'This is a pre-release'
```
Comment (johnnyaug) [inline release-checklist.md:16]:
```suggestion
- [ ] Uncheck 'This is a pre-release'
```

**Review by ozkatz — APPROVED** [review]
LGTM!

**Pull request merged**